crates/sui-indexer-alt-graphql/src/api/types/transaction_kind/programmable/commands/mod.rs

```diff
@@ -48,47 +48,51 @@ pub struct OtherCommand {
 }
 
 impl Command {
-    pub fn from(scope: Scope, command: NativeCommand) -> Self {
+    pub fn from(scope: Scope, command: NativeCommand) -> Result<Self, anyhow::Error> {
         match command {
-            NativeCommand::MakeMoveVec(_type_opt, elements) => {
-                Command::MakeMoveVec(MakeMoveVecCommand {
+            NativeCommand::MakeMoveVec(type_opt, elements) => {
+                Ok(Command::MakeMoveVec(MakeMoveVecCommand {
+                    type_: match type_opt {
+                        Some(type_input) => Some(MoveType::from_native(type_input.to_type_tag()?, scope.clone())),
```

Comment:
You should expose a new `MoveType::from_input` that constructs a `MoveType` directly from a `TypeInput` and use that here (because that's what `MoveType` uses as its internal representation).

We may need to represent a transaction where this type is incorrect in case we're presenting a transaction that failed to execute/simulate, so in that case we shouldn't fail here.